mutations/loadLock.js

```diff
@@ -70,117 +113,189 @@ async function loadLock(inputs, models, req) {
+        if (LockSearch.Min_Fakes === undefined || LockSearch.Max_Fakes === undefined) { // should never happen, but...
+            throw new Error("CreatedLock object allows fakes, but doesn't specify minimum and maximum fakes")
+        }
+        // LockSearch has min and max set if we got here
+        if (minFakes < LockSearch.Min_Fakes) {
+            validationErrors.push("Minimum fakes is less than lock's required minimum")
+        }
+        // tested
+        if (maxFakes > LockSearch.Max_Fakes) { 
+            validationErrors.push("Maximum fakes is more than lock's allowed maximum")
+        }
+        // tested
+    } else { //fakes are not allowed
+        if (minFakes > 0 || maxFakes > 0) {
+            validationErrors.push("The requested lock does not allow fake locks")
+        }
+        // tested
+    }
 
-       if(LockSearch.Limit_Users === true) {
-           const UserLimit = LockSearch.User_Limit_Amount; 
-           const NumOfLocksLoaded = models.LoadedLock.findAll({
-                where: {
-                    CreatedLock_ID: LockSearch.Lock_ID,
-                    $and: {Unlocked: false}
+    if (inputs.Emergency_Keys === true) { // no validation needed if false
+        if(LockSearch.Allow_Buyout === false) {
+            validationErrors.push("The keyholder does not allow emergency keys on this lock");
+            // tested
+            // no need to validate inputs.Emergency_Keys_Amount
+        } else { //lock allows buyout
+            if (inputs.Emergency_Keys_Amount === undefined) {
+                validationErrors.push("Number of keys must be provided if emerg. keys enabled.");
+                // tested
+            } else {
+                if (inputs.Emergency_Keys_Amount < 1 ) {
+                    validationErrors.push ("Number of keys must be at least one if emerg. keys enabled");
+                    //tested
                 }
-            })
+            }
+        }
+    } else {
+        if (inputs.Emergency_Keys_Amount != 0) { // anything other than zero or undefined
+            validationErrors.push("Number of emerg. keys specified without emerg. keys being enabled")
+            //tested
+        }
+    }
 
-            if(NumOfLocksLoaded.length >= UserLimit) {
-                validationErrors.push("The keyholder has set a limit on how many of these locks that can be loaded and the limit has been reached.");
+    if(LockSearch.Limit_Users === true) {
+        const UserLimit = LockSearch.User_Limit_Amount // should be defined, but not enforced by DB/model
+        if (UserLimit === undefined) { // should never happen, but...
+            throw new Error("CreatedLock object limits users, but doesn't specify limit")
+        }
+        const NumOfLocksLoaded = await models.LoadedLock.findAll({
+            where: {
+                CreatedLock_ID: LockSearch.Lock_ID,
+                Unlocked: false  // default is AND, $ syntax didn't work
```

Comment:
If the lock has been deleted from the database, then it won't be found by this code.  When we eventually write a mutator to allow a user to delete a lock in the app, we'll have to figure out whether that should it remove it from the database, If so, then we won't be able to include it in statistics for the lockee or keyholder, so I think not.  We might need to include another field to take care of abandoned locks (or revise Unlocked to something other than Boolean to support possibilities other than locked or unlocked.)  Alternatively, when a user deletes a lock in the app, summary information about the lock could be written into a different table for use in history and statistics.

Reply:
Exactly! Let's save this up for later.
About lock deletion, we certainly don't want to delete it from the DB, in case e.g. the user forgets the code or deletes it by accident, so it can be restored. And also for the stats.